shell/plugins/rancher-api/base-cluster-class.ts

```diff
@@ -0,0 +1,135 @@
+   */
+  protected canManageResource(type: string, schema: Schema, action: 'create' | 'edit' | 'delete'): Boolean {
+    let method: string;
+    let isAllowed: boolean | undefined;
+
+    switch ( action ) {
+    case 'create':
+      method = 'post';
+      isAllowed = schema?.collectionMethods?.some((x: string) => x.toLowerCase() === method);
+      break;
+    case 'edit':
+      method = 'patch';
+      isAllowed = schema?.resourceMethods?.some((x: string) => x.toLowerCase() === method);
+      break;
+    case 'delete':
+      method = 'delete';
+      isAllowed = schema?.resourceMethods?.some((x: string) => x.toLowerCase() === method);
+      break;
+    default:
+      throw new Error('Invalid action specified');
+    }
+
+    if ( !isAllowed ) {
+      return false;
+    }
+
+    switch ( action ) {
+    case 'create':
+      return this.$store.getters['type-map/optionsFor'](type).isCreatable;
+    case 'edit':
+      return this.$store.getters['type-map/optionsFor'](type).isEditable;
+    case 'delete':
+      return this.$store.getters['type-map/optionsFor'](type).isRemovable;
+    }
+  }
+
+  /**
+   * Helper function to build dispatch parameters for Vuex store actions.
+   *
+   * @param type - The resource schema ID.
+   * @param options - Additional options for fetching resources.
+   * @returns An object containing dispatch parameters.
+   */
+  private buildDispatchParams(type: string, options: ResourceFetchOptions = {}) {
+    const params: { [key: string]: any } = { type };
+
+    for (const [key, value] of Object.entries(options)) {
+      params[key] = value;
+    }
+
+    return params;
+  }
+
+  /**
+   * Fetch a resource type to return either a full list of resources by type, a singular resource
+   * by type and ID, or a list of resources by type that match specific label criteria defined in the
+   * selector. This is useful for filtering resources based on their metadata labels.
+   * The `force` option can be used to initiate a new dispatch request regardless of any existing cache.
+   *
+   * @param type The resource schema ID.
+   * @param options? Additional fetching options: `id?`, `selector?`, `namespace?`, `force?`.
+   *  - `id`: Specify the ID of the resource for fetching a single item.
+   *  - `selector`: A string used to match resources based on their labels.
+   *                The format is 'key=value', where 'key' is the label name and 'value' is the label value.
+   *                For example, 'app=kubewarden-policy-server-default' would match resources labeled with
+   *                'app' as 'kubewarden-policy-server-default'.
+   *  - `namespace`: Specify the namespace of the resource (if applicable).
+   *  - `force`: Force a new dispatch request, bypassing any cached data.
+   *
+   * @returns `Promise<any>` A promise that resolves with the fetched resource(s).
+   */
+  async get(type: string, options?: ResourceFetchOptions): Promise<any> {
```

Comment:
It's not obvious to me by looking at this signature how I should retrieve all items of a resource (from looking at the code it's obvious) which I think is problematic. Maybe changing options to filter instead would make that clear but I suspect we'll want to leave it as options to support pagination in the future.

I think I'd be in favor of splitting the functionality of findAll and findMatching/find into two different methods. If there's a better idea for making the usage obvious I'd be fine with something else.

Reply:
I was trying to emulate how running a `kubectl get <resource>` would return all of a resource (sans the namespace), maybe this isn't the best approach. I figured having Intellisense that explains the design of the function and options in detail would provide clear instructions on how to use the method.

I'm open to splitting the functionality as well, this may help limit confusion with further implementation.